src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -368,7 +368,13 @@ var DOMRenderer = ReactFiberReconciler({
       const hostContextDev = ((hostContext: any): HostContextDev);
       validateDOMNesting(null, text, null, hostContextDev.ancestorInfo);
     }
-    var textNode: TextInstance = document.createTextNode(text);
+
+    var doc = rootContainerInstance;
+    if (rootContainerInstance.nodeType !== DOCUMENT_NODE && doc.ownerDocument) {
```

Comment:
Why test for `doc.ownerDocument`? I don't know why it would ever be falsy and if it is then your could would break, because you would be trying to call `createTextNode` on some random element.

Reply:
We don't have to, I guess from the same flow typing info, it can only be an Element (which HAS ownerDocument) or it's a Document which is the document itself.